cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```
@@ -106,6 +110,34 @@ protected final void stopScheduler() throws SchedulerException {
   }
 
   @Override
+  public void addProgramSchedule(ProgramSchedule schedule) throws AlreadyExistsException, SchedulerException {
+    Scheduler scheduler = getScheduler(schedule);
+    if (scheduler == null) {
+      return;
```

is that not an error? Silently ignoring it does not seem right